IPython/core/debugger.py

```diff
@@ -324,15 +324,20 @@ def new_do_restart(self, arg):
     def postloop(self):
         self.shell.set_completer_frame(None)
 
-    def print_stack_trace(self):
+    def print_stack_trace(self, context=None):
+        if not context: 
+            context = self.context
+
         try:
             for frame_lineno in self.stack:
-                self.print_stack_entry(frame_lineno, context = 5)
+                self.print_stack_entry(frame_lineno, context=context)
         except KeyboardInterrupt:
             pass
 
     def print_stack_entry(self,frame_lineno,prompt_prefix='\n-> ',
-                          context = 3):
+                          context=None):
+        if not context:
```

Comment:
is `context=0` valid? If so, all of these `if not context` need to be `if context is not None`